core/server/src/main/java/alluxio/worker/file/FileDataManager.java

```diff
@@ -176,32 +176,34 @@ public void lockBlocks(long fileId, List<Long> blockIds) throws IOException {
       if (mPersistingInProgressFiles.containsKey(fileId)) {
         throw new IOException("the file " + fileId + " is already being persisted");
       }
-      try {
-        // lock all the blocks to prevent any eviction
-        for (long blockId : blockIds) {
-          long lockId = mBlockWorker.lockBlock(Sessions.CHECKPOINT_SESSION_ID, blockId);
-          blockIdToLockId.put(blockId, lockId);
-        }
-      } catch (BlockDoesNotExistException e) {
-        errors.add(e);
-        // make sure all the locks are released
-        for (long lockId : blockIdToLockId.values()) {
-          try {
-            mBlockWorker.unlockBlock(lockId);
-          } catch (BlockDoesNotExistException bdnee) {
-            errors.add(bdnee);
-          }
+    }
+    try {
+      // lock all the blocks to prevent any eviction
+      for (long blockId : blockIds) {
```

Comment:
I don't think this can happen because 
1. The block lock (ClientRWLock) is a semaphore with size = 1000.
2. There is only one heartbeat thread.

What you said can happen only if we have 1000 (or more) threads running heartbeat  and all of them tries to persist the same set of files (size >= 1000) and form a cycle. This is not going to happen as I don't think it makes sense to have more than 1 thread to do heartbeat for now?